src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ArrayTrailingCommaCheck.java

```diff
@@ -29,9 +29,21 @@
  * </p>
  * <p>
  * Rationale: Putting this comma in make is easier to change the
- * order of the elements or add new elements on the end.
+ * order of the elements or add new elements on the end. Main benefit of a trailing
+ * comma is that when you add new entry to an array, no surrounding lines are changed.
  * </p>
  * <p>
+ * The check allows leaving out the comma at the end if either left
+ * or right curly braces are on the same line as the last element of the array.
```

Comment:
> . BTW we have RightCurlyCheck which is responsible for such logic.

@Vladlis Neither RightCurly or LeftCurly support `ARRAY_INIT` token which is what the brace is. This would have to be a new issue.

> We could require } to be on a new line.

Which would just duplicate work of other checks. Even in issue, original author said this check wasn't suppose to be used alone. https://github.com/checkstyle/checkstyle/issues/1509#issuecomment-139809732